ghcide/src/Development/IDE/Plugin/Completions.hs

```diff
@@ -106,6 +109,44 @@ dropListFromImportDecl iDecl = let
     f x = x
     in f <$> iDecl
 
+resolveCompletion :: IdeState -> PluginId -> CompletionItem -> LSP.LspM Config (Either ResponseError CompletionItem)
+resolveCompletion ide _ comp@CompletionItem{_detail,_documentation,_xdata}
+  | Just resolveData <- _xdata
+  , Success (uri, NameDetails mod occ) <- fromJSON resolveData
+  , Just file <- uriToNormalizedFilePath $ toNormalizedUri uri = do
+    mdkm <- liftIO $ runAction "Completion resolve" ide $ use GetDocMap file
+    case mdkm of
+      Nothing -> pure (Right comp)
+      Just (DKMap dm km) -> liftIO $ runAction "Completion resolve" ide $ do
```

Comment:
I think you can make this kind of pattern simpler using the pattern we use elsewhere in the codebase using `pluginResponse` at the top level and `maybeToEither` and friends.